protoc-gen-kotlin/lib/src/commonMain/kotlin/pbandk/gen/CodeGenerator.kt

```diff
@@ -26,7 +30,35 @@ open class CodeGenerator(
         fn().also { indent = indent.dropLast(4) }
     }
 
-    protected fun writeType(type: File.Type, parentType: String? = null) = when (type) {
+    private fun writeExtension(field: File.Field) {
+        when (field) {
+            is File.Field.Numbered -> {
+                val localType = when (field.type) {
+                    File.Field.Type.MESSAGE -> ((field as File.Field.Numbered.Standard).localTypeName ?: field.kotlinLocalTypeName)?.substringAfter(".")
+                    else -> field.type.standardTypeName
```

Comment:
Instead of this `when` statement to calculate the value of `localType`, I _think_ you should be able to use `field.kotlinValueType(true)` as the value of `localType`.